### ISSUE TITLE
EBS volumes doesn't get details from BlockDeviceMappings

### ISSUE THREAD

**Issue by huynguyent** [spulec/moto#3219]
I tried to create a new EC2 instance through `ec2.run_instances()` and attach it to a newly created EBS. When I verify the EBS details through `ec2.describe_volumes()`, the details don't match what I specify in `BlockDeviceMappings` config at all.

```import boto3
from moto import mock_ec2
from moto.ec2 import ec2_backends

region = "us-west-1"

mock = mock_ec2()
mock.start()

kwargs = {
    "ImageId": "my_ami",
    "MaxCount": 1,
    "MinCount": 1,
    "KeyName": "my_key",
    "InstanceType": "my_instance_type",
    "EbsOptimized": False,
    "NetworkInterfaces": [
        {
            "DeviceIndex": 0,
            "Description": "Primary network interface",
            "DeleteOnTermination": True,
            "SubnetId": next(ec2_backends[region].get_all_subnets()).id,
            "Ipv6AddressCount": 0,
            "Groups": ["group"],
        }
    ],
    "BlockDeviceMappings": [
        {
            "DeviceName": "device",
            "Ebs": {"VolumeSize": 100, "DeleteOnTermination": True, "VolumeType": "gp2", 'Encrypted': True},
        }
    ],
}

ec2_client = boto3.client("ec2", region_name=region)

ec2_client.run_instances(**kwargs)
volumes = ec2_client.describe_volumes()

print(volumes)

assert volumes["Volumes"][0]["Attachments"][0]["Device"] == "device"
assert volumes["Volumes"][0]["Attachments"][0]["DeleteOnTermination"] is True
assert volumes["Volumes"][0]["Attachments"][0]["VolumeType"] == "gp2"
assert volumes["Volumes"][0]["Attachments"][0]["Encrypted"] is True

```
Here is the output of `ec2.describe_volumes()`
```{
    'Volumes': [
        {
            'Attachments': [
                {
                    'AttachTime': datetime.datetime(2020, 8,7 ,7,21, 30, tzinfo=tzutc()), 
                    'Device': '/dev/sda1', 
                    'InstanceId': 'i-87981627e45359299', 
                    'State': 'attached', 
                    'VolumeId': 'vol-061012f9', 
                    'DeleteOnTermination': False
                }
            ], 
            'AvailabilityZone': '', 
            'CreateTime': datetime.datetime(2020, 8, 7, 7, 21, 30, tzinfo=tzutc()), 
            'Encrypted': False, 
            'Size': 8, 
            'SnapshotId': '', 
            'State': 'in-use', 
            'VolumeId': 'vol-061012f9', 
            'VolumeType': 'standard'
        }
    ], 
    'ResponseMetadata': {
        'RequestId': '59dbff89-35bd-4eac-99ed-be587EXAMPLE', 
        'HTTPStatusCode': 200, 
        'HTTPHeaders': {
            'server': 'amazon.com'
        }, 
        'RetryAttempts': 0
    }
}
```

**Comment by bblommers** [comment]
Hi @huynguyent, which version are you on? This should be fixed as of moto >= 1.3.15.dev843 

PR with the fix: https://github.com/spulec/moto/pull/2949

**Comment by huynguyent** [comment]
@bblommers I was using 1.3.14. I'll give 1.3.15.dev843 a try. Thanks!

**Comment by davido-bs** [comment]
I'm on 1.3.14, but I also tried this on the 1.3.15 master branch without any help. I noticed that instances created from an autoscaling group also only show their root volume in the BlockDeviceMappings, even if an extra block device is specified in the launch configuration.

```python
@pytest.fixture
def credentials():
    os.environ["AWS_ACCESS_KEY_ID"] = "testing"
    os.environ["AWS_SECRET_ACCESS_KEY"] = "testing"
    os.environ["AWS_SECURITY_TOKEN"] = "testing"
    os.environ["AWS_SESSION_TOKEN"] = "testing"
    os.environ["AWS_DEFAULT_REGION"] = "us-east-1"


@pytest.fixture
def mock_ec2(credentials):
    with moto.mock_ec2():
        yield boto3.client("ec2")


@pytest.fixture
def mock_autoscaling(credentials):
    with moto.mock_autoscaling():
        yield boto3.client("autoscaling")


@pytest.fixture()
def random_image_id(mock_ec2):
    return mock_ec2.describe_images()["Images"][0]["ImageId"]


@pytest.fixture
def default_launch_config(mock_autoscaling, random_image_id):
    return mock_autoscaling.create_launch_configuration(
        LaunchConfigurationName=f"lc-{random_image_id}",
        ImageId=random_image_id,
        BlockDeviceMappings=[
            {
                "DeviceName": "/dev/sdf",
                "Ebs": {
                    "VolumeSize": 10,
                    "VolumeType": "standard",
                    "Encrypted": False,
                    "DeleteOnTermination": False,
                },
            }
        ],
    )

def create_asg(mock_autoscaling, random_image_id, tags):
    asg_name = f"asg-{random_image_id}"
    asg = mock_autoscaling.create_auto_scaling_group(
        AutoScalingGroupName=asg_name,
        LaunchConfigurationName=f"lc-{random_image_id}",
        MinSize=1,
        MaxSize=1,
        DesiredCapacity=1,
        AvailabilityZones=[DEFAULT_AZ],
        Tags=[
            {
                "ResourceId": asg_name,
                "ResourceType": "auto-scaling-group",
                "Key": key,
                "Value": value,
                "PropagateAtLaunch": True,
            }
            for key, value in tags.items()
        ],
    )
    return asg

@pytest.fixture
def default_asg(mock_autoscaling, random_image_id, default_launch_config):
    tags = DEFAULT_TAGS
    return create_asg(mock_autoscaling, random_image_id, tags)

def test_default_tags(mock_ec2, mock_autoscaling, default_asg):
    assert len(mock_ec2.describe_volumes()['Volumes']) == 2
```
The call to describe_volumes() only includes the default root EBS volume, not the extra one specified in the launch configurations' BlockDeviceMappings.

**Comment by bblommers** [comment]
Hi @davido-bs, the creation of volumes defined in launch configurations is now fixed with #5050, part of moto >= 3.1.6.dev13. I'll close this, but please let us know if  you're running into any other issues.
